src/components/Dashboard.tsx

```diff
@@ -0,0 +1,160 @@
+import useListTemplates from '../utils/hooks/useListTemplates'
+import usePageTitle from '../utils/hooks/usePageTitle'
+import { TemplateDetails, TemplateInList } from '../utils/types'
+
+const Dashboard: React.FC = () => {
+  const {
+    userState: { templatePermissions, isNonRegistered },
+    logout,
+  } = useUserState()
+  const {
+    templatesData: { templatesByCategory },
+  } = useListTemplates(templatePermissions, false)
+  if (isNonRegistered) {
+    logout()
+    return null
+  }
+
+  usePageTitle(strings.PAGE_TITLE_HOME)
+
+  return (
+    <div id="dashboard">
+      {templatesByCategory.map(
+        ({ templates, templateCategory: { categoryIcon, categoryTitle } }) => (
+          <div key={categoryTitle} className="template-category">
+            <div className="title">
+              {categoryIcon && (
+                <Icon size="large" color="grey" name={categoryIcon as SemanticICONS} />
+              )}
+              <Header as="h4">{categoryTitle}</Header>
+            </div>
+            <div className="templates">
+              {templates.map((template) => (
+                <TemplateComponent key={template.code} template={template} />
+              ))}
+            </div>
+          </div>
+        )
+      )}
+    </div>
+  )
+}
+
+const TemplateComponent: React.FC<{ template: TemplateInList }> = ({ template }) => {
+  const { name, code, hasApplyPermission, hasFilters, filters } = template
+  return (
+    <div className="template">
+      <div className="title">
+        <Header as="h5">{name}</Header>
+
+        {hasApplyPermission && (
+          <Button inverted size="small" as={Link} to={`/application/new?type=${code}`} primary>
+            {constants.BUTTON_DASHBOARD_NEW}
+          </Button>
+        )}
+      </div>
+      {hasFilters && (
+        <div className="filters">
+          {filters.map((filter) => (
+            <FilterComponent key={filter.id} template={template} filter={filter} />
+          ))}
+        </div>
+      )}
+      <ViewAll template={template} />
+    </div>
+  )
+}
+
+const FilterComponent: React.FC<{ template: TemplateDetails; filter: Filter }> = ({
+  template,
+  filter,
+}) => {
+  const templateType = template.code
+  const { loading, applicationCount } = useListApplications({
+    type: templateType,
+    perPage: 1,
+    ...filter.query,
+  })
+
+  const applicationListUserRole =
+    filter.userRole === PermissionPolicyType.Apply ? USER_ROLES.APPLICANT : USER_ROLES.REVIEWER
+
+  const constructLink = () =>
+    `/applications?type=${templateType}&user-role=${applicationListUserRole}&${Object.entries(
+      filter.query
+    )
+      .map(([key, value]) => `${key}=${value}`)
+      .join('&')}`
+
+  if (loading) return null
+  if (applicationCount === 0) return null
+
+  return (
+    <div className="filter">
+      <Link to={constructLink()}>
+        {filter.icon && (
+          <Icon color={filter.iconColor as SemanticCOLORS} name={filter.icon as SemanticICONS} />
```

Comment:
Wonder why we need the `as` here, haven't had to use it before. 🤷‍♂️

Reply:
.iconColor and .icon are both strings, need to cast to semanticICON (make it that type and cast in the hook too, not sure where is the best place)